Octokit.Tests/Clients/RepositoryContentsClientTests.cs

```diff
@@ -68,35 +68,394 @@ public class TheGetReadmeHtmlMethod
+                };
+
+                var connection = Substitute.For<IApiConnection>();
+                connection.GetAll<RepositoryContent>(Args.Uri, options).Returns(Task.FromResult(result.AsReadOnly() as IReadOnlyList<RepositoryContent>));
+                var contentsClient = new RepositoryContentsClient(connection);
+
+                var contents = await contentsClient.GetAllContentsByRef("fake", "repo", "master", options);
+
+                connection.Received().GetAll<RepositoryContent>(Arg.Is<Uri>(u => u.ToString() == "repos/fake/repo/contents/?ref=master"), options);
+                Assert.Equal(1, contents.Count);
+            }
+
+            [Fact]
+            public async Task ReturnsContentsByRefWithPath()
+            {
+                var result = new List<RepositoryContent> { new RepositoryContent() };
+
+                var connection = Substitute.For<IApiConnection>();
+                connection.GetAll<RepositoryContent>(Args.Uri, Args.ApiOptions).Returns(Task.FromResult(result.AsReadOnly() as IReadOnlyList<RepositoryContent>));
+                var contentsClient = new RepositoryContentsClient(connection);
+
+                var contents = await contentsClient.GetAllContentsByRef("fake", "repo", "readme.md", "master");
+
+                connection.Received().GetAll<RepositoryContent>(Arg.Is<Uri>(u => u.ToString() == "repos/fake/repo/contents/readme.md?ref=master"), Args.ApiOptions);
                 Assert.Equal(1, contents.Count);
             }
+
+            [Fact]
+            public async Task ReturnsContentsByRefWithPathAndApiOptions()
+            {
+                var result = new List<RepositoryContent> { new RepositoryContent() };
+
+                var options = new ApiOptions
+                {
+                    StartPage = 1,
+                    PageCount = 1,
+                    PageSize = 1
+                };
+
+                var connection = Substitute.For<IApiConnection>();
+                connection.GetAll<RepositoryContent>(Args.Uri, options).Returns(Task.FromResult(result.AsReadOnly() as IReadOnlyList<RepositoryContent>));
+                var contentsClient = new RepositoryContentsClient(connection);
+
+                var contents = await contentsClient.GetAllContentsByRef("fake", "repo", "readme.md", "master", options);
+
+                connection.Received().GetAll<RepositoryContent>(Arg.Is<Uri>(u => u.ToString() == "repos/fake/repo/contents/readme.md?ref=master"), options);
+                Assert.Equal(1, contents.Count);
+            }
+
+            [Fact]
+            public async Task EnsuresNonNullarguments()
+            {
+                var connection = Substitute.For<IApiConnection>();
+                var client = new RepositoryContentsClient(connection);
+
+                // public Task<IReadOnlyList<RepositoryContent>> GetAllContents(string owner, string name)
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAllContents(null, null));
```

Comment:
For the null checks dont we just need to check each argument is null, not also checking multiple arguments null at the same time?  So in this function all the ones with more than one argument being null at the same time, can be removed I think